server/src/com/thoughtworks/go/config/update/AgentsEntityConfigUpdateCommand.java

```diff
@@ -141,26 +129,94 @@ public void update(CruiseConfig preprocessedConfig) throws Exception {
         }
     }
 
-    private void checkElasticAgentsResourceUpdated(List<AgentConfig> uuids, List<String> resourcesToAdd, List<String> resourcesToRemove) throws ElasticAgentsResourceUpdateException {
-        ArrayList<String> elasticAgentUUIDs = findAllElasticAgentUuids(uuids);
+    private void validateAgentUUIDs() throws NoSuchAgentException {
```

Comment:
We can have a generic exception ```NoSuchResourceException``` instead of creating an exception class for each resource.